### PR TITLE
Support JARs as test sources in Quarkus JUnit 5 extension

### LINKED ISSUE #2941: Support JARs as test sources in Quarkus JUnit 5 extension
Currently, Quarkus test extension expects the tests to be located in a directory on the disk. The testsuites of the externally built Quarkus extensions, however, will be integrated into the platform testsuite as JAR files. This issue is about adding support for JAR files as test sources to the Quarkus JUnit5 extension.

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #2941

### CODE REVIEW

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java [hunk 1]
```diff
@@ -279,6 +300,7 @@ public Object createTestInstance(TestInstanceFactoryContext factoryContext, Exte
                     try {
                         launcher.start();
                     } catch (IOException e) {
+                        launcher.close();
```
Comment (emmanuelbernard) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java]: What if close fails? :) I'm ok with that change but is that with an understanding that launcher.close() will always work even of start failed?

**Comment by emmanuelbernard** [comment]
Looks good, just added a minor comment

**Comment by emmanuelbernard** [comment]
BTW @cescoffier / @gsmet if TCK fails, do we deny a PR? Or is that a non mandatory step?

**Review by emmanuelbernard — APPROVED** [review]

**Comment by aloubyansky** [comment]
yes, i'm looking into that failure.

**Comment by gsmet** [comment]
@emmanuelbernard in this case, I think we need to fix it as it's an infrastructure issue. Otherwise, it's on a case per case basis, we already have a few tests ignored.

**Comment by aloubyansky** [comment]
Fixed it.

**Comment by aloubyansky** [comment]
There was hacky workaround in the QuarkusDeployableContainer to make tests from JARs run, which I didn't notice before. So it fits right into this issue.

File: test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java [hunk 2]
```diff
@@ -45,8 +62,23 @@ public static Path getTestClassesLocation(Class<?> testClass) {
...
+     * Resolves the directory or the JAR file containing the application being tested by the test class.
+     *
+     * @param testClass the test class
+     * @return directory or JAR containing the application being tested by the test class
+     */
     public static Path getAppClassLocation(Class<?> testClass) {
-        String testClassPath = getTestClassesLocation(testClass).toString();
+        final String testClassPath = getTestClassesLocation(testClass).toString();
+        if (testClassPath.endsWith(".jar")) {
+            if (testClassPath.endsWith("-tests.jar")) {
```
Comment (stuartwdouglas) [inline test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java]: I think at some point we need to revisit this and provide some way for the execution environment to pass in this information, so it is not hard coded like this. This is fine for the moment though.

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**